ide/coqide/session.ml

```diff
@@ -222,9 +224,19 @@ let set_buffer_handlers
       (* If coq was asked to backtrack, the cleanup must be done by the
          backtrack_until function, since it may move the stop_of_input
          to a point indicated by coq. *)
-      let iters = List.map (fun mark -> buffer#get_iter_at_mark (`MARK mark)) marks in
+      let iters = List.map (fun mark ->
+          match mark with
+          | `Mark mark -> buffer#get_iter_at_mark (`MARK mark)
+          | `Begin -> buffer#start_iter
+        ) marks in
       let iter = List.hd @@ List.sort (fun it1 it2 -> it1#compare it2) iters in
-      let () = List.iter (fun mark -> try buffer#delete_mark (`MARK mark) with GText.No_such_mark _ -> ()) marks in
+      let () = List.iter (fun mark ->
+        try match mark with
+        | `Mark mark -> buffer#delete_mark (`MARK mark)
+        | `Begin -> let action = Coq.seq (coqops#backtrack_to_begin ())
```

Comment:
Why don't you simply use `handle_reset_initial` there? I think it's precisely behaving like this piece of code.

Reply:
Calling `handle_reset_initial` gives the error "Already initialized" from `Idetop.init`.  Calling `Coq.reset_coqtop` works, but it doesn't correctly clear breakpoints in the replaced part of the buffer.  I'd rather not tie this code to breakpoints here, at least for now.